### PR TITLE
Problem: (0.6.x)tendermint could include duplicated tx which mess up the tx indexer

### PR DESCRIPTION
**Pull request by yihuang**
ref: #523

Solution:
- update gomod2nix
- update tendermint dependency to include the fix

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -169,3 +172,5 @@ replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0
 // Note: gorocksdb bindings for OptimisticTransactionDB are not merged upstream, so we use a fork
 // See https://github.com/tecbot/gorocksdb/pull/216
 replace github.com/tecbot/gorocksdb => github.com/cosmos/gorocksdb v1.1.1
+
+replace github.com/tendermint/tendermint => github.com/yihuang/tendermint v0.34.14-0.20220527100458-1338c66f8f74
```
Comment (yihuang) [inline go.mod:176]: TODO: remove replace after https://github.com/tendermint/tendermint/pull/8625 merged.

File: go.mod [hunk 2]
```diff
@@ -161,11 +165,12 @@ replace github.com/cosmos/iavl => github.com/cosmos/iavl v0.17.3
...
+replace github.com/cosmos/cosmos-sdk => github.com/crypto-org-chain/cosmos-sdk v0.44.6-cronos-revert-3
 
-replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.7.2-cronos-15
+replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.7.2-cronos-16
 
 // Note: gorocksdb bindings for OptimisticTransactionDB are not merged upstream, so we use a fork
 // See https://github.com/tecbot/gorocksdb/pull/216
 replace github.com/tecbot/gorocksdb => github.com/cosmos/gorocksdb v1.1.1
+
+replace github.com/tendermint/tendermint => github.com/crypto-org-chain/tendermint v0.34.20-0.20220630092134-c8c4a9fc4851
```
Comment (yihuang) [inline go.mod:176]: https://github.com/tendermint/tendermint/compare/v0.34.19...crypto-org-chain:tendermint:fix-duplicated-tx-index

Comment (thomas-nguy) [reply go.mod]: does this bug also 0.6.x?

why not just patch tendermint v0.34.14 with the targeted commit (c8c4a9fc4851d4e49b1576c8071386b09ad1d2f5) ?

Comment (yihuang) [reply go.mod]: yes, after the fix, the user doesn't need to patch the tx indexer after sync.

Comment (yihuang) [reply go.mod]: > why not just patch tendermint v0.34.14 with the targeted commit

it's better to keep dependencies up to date?

**Review by thomas-nguy — APPROVED** [review]
lgtm. 

I did a quick check on the diff and it should be add any breaking changes but better to test before releasing

**Pull request merged**